### DISCUSSION TITLE
Genesis – Haxe/OpenFL app integration

### DISCUSSION THREAD

**Discussion by Igazine** [prominic/Moonshine-IDE#1114] [Ideas]
### Description

Add Haxe multi-platform experiences to Genesis that build "out of the box" with Moonshine.

### Main Goals

One-click download Haxe/OpenFL apps (source code and project files) from Genesis directory that open (and possibly build and run) in Moonshine (similarly to currently available Domino apps)

### Optional (good-to-have) Goals

One-click build for multiple target platforms

### Build Targets

* macOS (native)
* Windows (native)
* HTML5/JS
* iOS and iPadOS
* Android

The discussion is open for anyone, tagging a few users:
@Igazine 
@JoelProminic 
@rat-moonshine 
@piotrzarzycki21

**Comment by rat-moonshine** [comment]
@Igazine Thank you, for creating the topic. Maybe some description on what we want to achieve, would be helpful for everyone.

**Comment by Igazine** [comment]
## Haxe/OpenFL app configuration options

### Compilation flags, conditional compilation, Haxedef

* [Compilation flags](https://haxe.org/manual/lf-condition-compilation.html) can be defined as Haxe/OpenFL build command line parameters

### project.xml

* By default OpenFL apps can be configured by customizing the [project.xml](https://lime.openfl.org/docs/project-files/xml-format/) file.
* A lot of things can be defined/customized in the XML based on build targets, platforms, build environment, conditionals etc.
* This file can be generated on the server on-the-fly, and pre-filled with values based on the current user etc. if needed before the zip is generated

### Custom XML file

* Additionally custom XML files can be defined and used for different build scenarios, and can be defined as OpenFL build parameters (e.g. `project-mac.xml`, `project-ios.xml` etc)

**Comment by Igazine** [comment]
## Demo App

The demo Haxe/OpenFL app (with a few basic functions) for Genesis is currently a work-in-progress.
